File: frontend/src/types/platform.ts
/**
 * 平台配置类型定义
 * ------------------------------------------------
 * 前端是通用"壳"，所有平台配置由后端 GET /api/platforms 返回。
 * 前端通过这些类型定义来解析配置，动态注册路由和渲染页面。
 */

/** 导航菜单项 */
export interface NavItem {
  /** 路由路径，如 /aluminum/overview（一级菜单有 path，二级菜单也直接用完整路径） */
  path?: string
  /** 菜单显示名称 */
  label: string
  /** Element Plus 图标组件名，如 Odometer、Grid、Bell */
  icon?: string
  /** 子菜单（二级菜单），有 children 时 path 可省略（作为分组标题不跳转） */
  children?: NavItem[]
}

/** 表格列定义 */
export interface ColumnDef {
  /** 数据字段名 */
  field: string
  /** 列标题 */
  label: string
  /** 列宽度 */
  width?: number
  /** 对齐方式 */
  align?: 'left' | 'center' | 'right'
  /** 列渲染类型 */
  type?: 'text' | 'tag' | 'dot' | 'temperature' | 'number'
  /** 单位（如 ℃、kW、V） */
  unit?: string
  /** tag/dot 类型的选项映射，key 为原始值，value 为显示文本 */
  options?: Record<string, string>
}

/** 页面配置 */
export interface PageConfig {
  /** 页面类型：表格 / 仪表盘 / 图表 / 自定义组件 */
  type: 'table' | 'dashboard' | 'chart' | 'custom'
  /** 页面标题 */
  title: string
  /** table 类型的列定义 */
  columns?: ColumnDef[]
  /** 数据接口路径（相对于 /api/:platform/），如 devices、alarms */
  api?: string
  /** chart 类型的图表类型：line / bar / pie */
  chartType?: string
  /** 仪表盘/图表的布局配置 */
  layout?: any
  /** custom 类型：自定义组件名称（对应组件注册表中的 key） */
  component?: string
}

/** 平台配置（内嵌在 Platform 的 config 字段中） */
export interface PlatformConfig {
  /** 导航菜单项列表 */
  navItems: NavItem[]
  /** 页面配置映射，key 为页面路径标识（如 overview、devices） */
  pages: Record<string, PageConfig>
}

/** 平台定义 —— 后端返回的完整平台对象 */
export interface Platform {
  /** 平台唯一标识，如 aluminum、pv */
  id: string
  /** 平台名称，如 铝厂云控平台 */
  name: string
  /** 平台图标（Emoji 或 Element Plus 图标名） */
  icon: string
  /** 平台数据库名/Schema */
  schema: string
  /** 平台配置（含菜单和页面定义） */
  config: PlatformConfig
  /** 平台状态：active / disabled */
  status: string
  /** 排序序号 */
  sortOrder: number
}

/** API 统一响应结构 */
export interface ApiResponse<T = any> {
  code: number
  message: string
  data: T
}

/** 分页响应结构 */
export interface PageResult<T = any> {
  list: T[]
  total: number
  page: number
  pageSize: number
}

/** 用户信息 */
export interface UserInfo {
  id: number
  username: string
  role: string
  platforms: string
  status: string
}

File: backend/internal/model/platform.go
package model

import (
	"encoding/json"
	"time"
)

// Platform 平台注册表模型，存储于 public schema。
// 每接入一个新平台，仅需在此表新增一条记录并创建对应 schema，后端自动路由。
type Platform struct {
	ID        string          `json:"id" gorm:"primaryKey"`                      // 平台唯一标识："aluminum", "pv", "water"
	Name      string          `json:"name"`                                      // 平台中文名："铝厂云控平台"
	Icon      string          `json:"icon"`                                      // 平台图标 emoji："🏭"
	Schema    string          `json:"schema" gorm:"column:schema_name"`          // 平台专属 schema 名："schema_aluminum"
	Config    json.RawMessage `json:"config" gorm:"type:jsonb"`                  // 平台 UI 配置(菜单/页面/字段定义)
	Status    string          `json:"status" gorm:"default:'active'"`            // active/inactive
	SortOrder int             `json:"sortOrder" gorm:"column:sort_order;default:0"`
	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
}

// TableName 指定表名
func (Platform) TableName() string {
	return "platforms"
}

// ---- 平台 UI 配置结构(用于构建/解析 Config JSON) ----
// 注意：这些结构必须与前端 types/platform.ts 中的定义保持一致。

// PlatformConfig 平台完整 UI 描述，包含导航菜单与页面定义
type PlatformConfig struct {
	NavItems []NavItem            `json:"navItems"`
	Pages    map[string]PageDef   `json:"pages"`
}

// NavItem 侧边导航菜单项
type NavItem struct {
	Path     string     `json:"path,omitempty"`     // 路由路径，如 /aluminum/devices（分组标题可省略）
	Label    string     `json:"label"`              // 显示名："设备列表"
	Icon     string     `json:"icon,omitempty"`     // 图标标识："Monitor"
	Children []NavItem  `json:"children,omitempty"` // 子菜单（二级菜单）
}

// PageDef 页面定义，前端据此动态渲染
type PageDef struct {
	Type      string      `json:"type"`              // 页面类型：table / chart / dashboard / custom
	Title     string      `json:"title"`             // 页面标题
	API       string      `json:"api"`               // 数据接口资源名(相对 /api/:platform/)，如 devices、alarms
	Columns   []ColumnDef `json:"columns,omitempty"` // 表格列定义(table 类型时使用)
	Component string      `json:"component,omitempty"` // 自定义组件名(custom 类型时使用)，如 AluminumCellMonitor
}

// ColumnDef 表格列定义
type ColumnDef struct {
	Field   string            `json:"field"`             // 数据字段名
	Label   string            `json:"label"`             // 列标题
	Type    string            `json:"type"`              // 渲染类型：text/tag/dot/temperature/number
	Width   int               `json:"width,omitempty"`   // 列宽(像素)
	Unit    string            `json:"unit,omitempty"`    // 单位(如 ℃、kW)
	Options map[string]string `json:"options,omitempty"` // tag/dot 类型的选项映射
}
